src/design-pages/passwordless_gdm.rst

```diff
@@ -0,0 +1,461 @@
+
+* GDM prompts the user for their username.
+
+* The user initiates the login procedure by entering their username.
+
+* GDM starts the PAM conversation.
+
+* GDM starts the authentication process in SSSD.
+
+* pam_sss communicates with the PAM responder to resolve the user.
+
+* PAM responder resolves the username and obtains, among other things, the
+  available authentication methods and the prompting strings.
+
+* PAM responder prepares the JSON message with the available authentication
+  mechanisms, the prompts and the user credentials to be requested
+  (check format in :ref:`data`).
+
+* PAM responder returns this information to pam_sss.
+
+* pam_sss wraps the JSON message in the PAM conversation using the macros
+  provided by GDM in ``/usr/include/gdm/gdm-custom-json-pam-extension.h``.
+
+* GDM prompts the user for the user credentials, and at the bottom of the
+  screen it shows the available authentication mechanisms. If the user selects
+  another method, GDM already has all the information needed to request the
+  user credentials for this authentication mechanism.
+ 
+* The authentication workflow continues, but varies depending on the selected
+  authentication mechanism. This will be explained separately for each
+  mechanism.
+
+.. _data:
+
+Data
+****
+In addition, the messages exchanged follow the JSON standard. SSSD provides the
+available authentication mechanisms in the following message:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "mechanisms": {
+          "$mech1": {
+            "name": "$name1",
+            "role": "$role1",
+            "selectable": true,
+            "msg1": "$msg1"
+          },
+          "$mech2": {
+            "name": "$name2",
+            "role": "$role2",
+            "selectable": true,
+            "msg1": "$msg2",
+            "msg2": "$msg3"
+          }
+        },
+        "priority": [
+          "$role2",
+          "$role1"
+        ]
+      }
+    }
+
+The field meaning is as follows:
+
+* name: the mechanism name that will be shown in the login screen.
+* role: the authentication mechanism. It can be password, EIdP, smartcard or
+  passkey.
+* selectable: whether the authentication mechanism should be shown up in the
+  login screen as an explicitly selectable option. By default it is set to
+  ``true``.
+
+GDM answers with the result for the previous JSON message processing:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "status": "$status",
+        "mech": {
+          "data1": "$data1",
+          "data2": "$data2"
+        }
+      }
+    }
+
+For clarification, there are two different types of user interaction messages:
+`prompt` and `instruction`. The first one asks the user to enter something on
+the screen, e.g. the PIN. The second type prompts the user to follow
+instructions, e.g. to enter the PIN into the reader.
+
+Examples for the two messages are defined in the following lines.
```

Comment:
Hi,

imo the wording is a bit confusing. In the paragraph before you say "...there are two different types of user interaction messages ..." and to me it sounds like "Examples for the two messages ..." refers to those. Additionally "Examples ... are defined" sounds a bit odd.

Maybe better say something like "Examples for the JSON request and reply messages are given in the following sections" or "The JSON request and reply messages for different authentication mechanisms are defined in the next sections."

bye,
Sumit

Reply:
I changed it to your last proposal